nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/service/RenditionServiceImpl.java

```diff
@@ -579,7 +588,31 @@ public Rendition getDefaultRendition(DocumentModel doc, String reason, Map<Strin
         } catch (ScriptException e) {
             log.error("Failed to evaluate script: ", e);
         }
+        if (store) {
+            StoredRendition storedRendition = storeRendition(doc, rendition);
+            if (storedRendition != null) {
+                return storedRendition;
+            }
+        }
         return rendition;
     }
 
+    @Override
+    public DocumentModel publishRendition(DocumentModel doc, DocumentModel target, String renditionName,
+            boolean override) {
+        RenditionService rs = Framework.getService(RenditionService.class);
```

Comment:
This is the RenditionService

Reply:
And?

Reply:
And you can drop `RenditionService rs = Framework.getService(RenditionService.class);`

Reply:
but of course :) tx